### PR TITLE
[rllib] Rename PolicyEvaluator => RolloutWorker

### LINKED ISSUE #4813: [rllib] [rfc] Proposed class renames
Per discussion with several users, there are some possible renames we could do to clarify the internal architecture:

- Rename `rllib.evaluation.PolicyGraph` to `rllib.policy.Policy`. Similarly, `rllib.evaluation.TFPolicyGraph` becomes `rllib.policy.TFPolicy` and so on. The files are moved to a new `rllib/policy` dir.

- ~Move the `rllib/agents` directory to `rllib/train`.~

- Add TF qualifier for consistency with Torch policies
    `A3CPolicyGraph => A3CTFPolicy`
    `A3CTorchPolicyGraph => A3CTorchPolicy`

- Rename `PolicyEvaluator` to `RolloutWorker`
    deprecate `compute_gradients()` / `apply_gradients()`

- Consolidate `[local_evaluator, remote_evaluators]` to single `WorkerSet` object
    ```
    workers = WorkerSet(...)
    local_evaluators => workers.local_worker()
    remote_evaluators => workers.remote_workers()
    ```

For the most part, I think we can do these renames without breaking backwards compatibility by leaving aliases behind. Though, I'm not sure how easy moving an entire directory is.

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

This finishes the renames started in https://github.com/ray-project/ray/pull/4819.

The documentation changes are separate in https://github.com/ray-project/ray/pull/4821

## Related issue number

Closes https://github.com/ray-project/ray/issues/4813

## Linter

- [x] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

File: doc/source/rllib-concepts.rst [hunk 1]
```diff
@@ -48,9 +48,9 @@ Most interaction with deep learning frameworks is isolated to the `Policy interf
...
 
-You can use policy evaluation standalone to produce batches of experiences. This can be done by calling ``ev.sample()`` on an evaluator instance, or ``ev.sample.remote()`` in parallel on evaluator instances created as Ray actors (see ``PolicyEvaluator.as_remote()``).
+You can use rollout workers standalone to produce batches of experiences. This can be done by calling ``worker.sample()`` on an worker instance, or ``worker.sample.remote()`` in parallel on worker instances created as Ray actors (see `WorkerSet <https://github.com/ray-project/ray/blob/master/python/ray/rllib/evaluation/worker_set.py>`__).
 
 Here is an example of creating a set of policy evaluation actors and using the to gather experiences in parallel. The trajectories are concatenated, the policy learns on the trajectory batch, and then we broadcast the policy weights to the evaluators for the next round of rollouts:
```
Comment (richardliaw) [inline doc/source/rllib-concepts.rst]: fix?

File: python/ray/rllib/evaluation/metrics.py [hunk 2]
```diff
@@ -39,27 +39,23 @@ def get_learner_stats(grad_info):
...
     metrics = summarize_episodes(episodes, episodes, num_dropped)
     return metrics
 
 
 @DeveloperAPI
-def collect_episodes(local_evaluator=None,
-                     remote_evaluators=[],
+def collect_episodes(local_worker=None, remote_workers=[],
                      timeout_seconds=180):
     """Gathers new episodes metrics tuples from the given evaluators."""
```
Comment (richardliaw) [inline python/ray/rllib/evaluation/metrics.py]: fix

File: python/ray/rllib/tests/test_optimizers.py [hunk 3]
```diff
@@ -11,7 +11,8 @@
 from ray.rllib.agents.ppo import PPOTrainer
 from ray.rllib.agents.ppo.ppo_policy import PPOTFPolicy
 from ray.rllib.evaluation import SampleBatch
-from ray.rllib.evaluation.policy_evaluator import PolicyEvaluator
+from ray.rllib.evaluation.rollout_worker import RolloutWorker
+from ray.rllib.evaluation.worker_set import WorkerSet
 from ray.rllib.optimizers import AsyncGradientsOptimizer, AsyncSamplesOptimizer
 from ray.rllib.optimizers.aso_tree_aggregator import TreeAggregator
 from ray.rllib.tests.mock_evaluator import _MockEvaluator
```
Comment (richardliaw) [inline python/ray/rllib/tests/test_optimizers.py]: can you rename this to MockWorker too?

File: doc/source/rllib-concepts.rst (hunk 1 above)
Comment (richardliaw) [reply doc/source/rllib-concepts.rst]: "policy weights to the evaluators "

**Comment by richardliaw** [comment]
Missed a few I think; can you make sure you grep for `evaluat*` (probably some docstrings missed too that weren't show on diff)

**Comment by richardliaw** [comment]
BTW, we talked about removing gradient calculation from the Evaluation abstraction; that's to be done later right?

File: doc/source/rllib-concepts.rst (hunk 1 above)
Comment (ericl) [reply doc/source/rllib-concepts.rst]: Fixed

**Comment by ericl** [comment]
Fixed.

**Comment by ericl** [comment]
Friendly ping on review.

**Review by richardliaw — APPROVED** [review]

**Pull request merged**